src/test/libslic3r/test_arc_fitting.cpp

```diff
@@ -0,0 +1,328 @@
+#include <catch.hpp>
+#include "libslic3r.h"
+#include "Point.hpp"
+#include "Geometry.hpp"
+#include "GCode/ArcFitting.hpp"
+#include "Log.hpp"
+#include "test_data.hpp"
+using namespace Slic3r;
+using namespace Slic3r::Test;
+
+void make_circle(Points &output, size_t n, double r, Point center = Point(0,0), double a = 0.0, double b = 2*PI, double noise = 0.0){
+    double arcdiff = (b-a)/n;
+    for(auto i = 0U; i < n; i++){
+        output.push_back(Point(r*cos(a)+center.x,r*sin(a))+center.y);
+        a += arcdiff;
+    }
+}
+
+SCENARIO("Arc detection works"){
+    GIVEN("A simple line"){
+        Polyline pl;
+        pl.points = {Point(5,15),Point(10,15),Point(15,15), Point(20,15), Point(25,15)};
+        THEN("No arcs are detected"){
+            auto arcs = arc_detection(pl, 4, 1.0);
+            REQUIRE(arcs.size() == pl.points.size());
+            for(auto &arc : arcs){
+                REQUIRE(arc == 0);
+            }
+        }
+    }
+    GIVEN("An arc with too few points"){
+        Polyline pl;
+        make_circle(pl.points,4,9000);
+        THEN("No arcs are detected"){
+            auto arcs = arc_detection(pl, 4, 2.0);
+            REQUIRE(arcs.size() == pl.points.size());
+            for(auto &arc : arcs){
+                REQUIRE(arc == 0);
+            }
+        }
+    }
+    GIVEN("A circle"){
```

Comment:
You should be more explicit here about the circle's parameters. Is it centered about a specific point in space? What's the expected radius in human terms?

Reply:
Ultimately, that test should pass with any circle (besides perhaps a 0 radius circle), so I didn't describe the specific parameters used.